File: coding-test/nodejs/src/practice/DetermineSquareRootOfAnInteger.js
// @ts-check

/**
 * @param {number} n
 * @returns {number}
 */
function solution(n) {
  const sqrtN = Math.sqrt(n)
  return Math.floor(sqrtN) === sqrtN ? Math.pow(sqrtN + 1, 2) : -1
}

const n = 121
console.log(solution(n))

File: practice-coding-test/java-kotlin/src/kr/co/programmers/java/practice/DetermineSquareRootOfAnInteger.java
package kr.co.programmers.java.practice;

public class DetermineSquareRootOfAnInteger {

    public static void main(String[] args) {
        long n = 121;
        System.out.println(new Solution().solution(n));
    }

    static class Solution {
        public long solution(long n) {
            if(Math.sqrt(n) - (int)Math.sqrt(n) > 0){
                return -1;
            }

            return (long) Math.pow(Math.sqrt(n) + 1, 2);
        }
    }
}
